---
title: "R Notebook"
output:
  pdf_document: default
  html_notebook: default
editor_options:
  chunk_output_type: console
---

## Clear workspace and load JAMP

```{r include=FALSE}
# Clear workspace and close all graphics devices --------------------------

rm(list = ls())
graphics.off()

library(JAMP)

```


## Set working directory to proper folder

```{r include=FALSE}

# set working directory to proper repo

setwd("/Users/jaredfreedman/Research/git/edna_metabarcoding/GC_VR_MiSeq_2021")

```

## Check for PhiX

```{r include=FALSE}
# check for PhiX

# only subsample 10000 reads

if(T){
  system2("vsearch", 
          "-fastx_subsample GC_VR_MiSeq_Demulti/Undetermined_S0_R1_001.fastq -fastaout GC_VR_MiSeq_Demulti/Undetermined_S0_R1_001.fastq -sample_pct 100")
} else { # check all reads in read 1 for PhiX
  system2("paste",
          " - - - - < /GC_VR_MiSeq_Demulti/Undetermined_S0_R1_001.fastq | cut -f 1,2 | sed 's/^@/>/' | tr \"\t\" \"\n\" > /GC_VR_MiSeq_Demulti/Undetermined_S0_R1_001.fastq")
}


system2("vsearch",
        "-usearch_global /GC_VR_MiSeq_Demulti/Undetermined_S0_R1_001.fastq -db /Users/jaredfreedman/Research/git/JAMP/Tutorial/PhiX.fasta -id 0.9 -strand both -blast6out PhiX_table.txt -maxrejects 1 -maxaccepts 1")
```

## Merge pair end reads

```{r}
demulti_files <- list.files("demulti_fastq/_data", full.names = TRUE)


Merge_PE(files = demulti_files)

#Successfully merged

```


## Trim primers
fwhF2 and EPTDr2n

EPTDr2n ended with a Y degenerate base, which IDT was unable to produce. Instead, two primers were created, one ending with C (EPTDr2n_C) and the other ending with T (EPTDr2n_T).

All primers are sets of 4 primers with 0-3 N bases to create frame shifts.

```{r include=FALSE}

# import primer sequences (just COI primer sequence, not including universal tail)

primers <- read.csv("Sites_and_Samples/fwhF2_EPTDr2n_primer_sequences.csv")

fwhF2 <- primers$Sequence[primers$Name=="fwhF2"]
EPTDr2n <- primers$Sequence[primers$Name=="EPTDr2n"]

# trim primers 

Cutadapt(forward = fwhF2,
         reverse = EPTDr2n,
         bothsides=T)

#by using "bothsides=T", forward or reverse primers are detected on both ends. This is not nessesary for fusion primers.
```









