---
title: "R Notebook"
output:
  pdf_document: default
  html_notebook: default
editor_options:
  chunk_output_type: console
---

## Clear workspace and load JAMP

```{r include=FALSE}
# Clear workspace and close all graphics devices --------------------------

rm(list = ls())
graphics.off()

library(JAMP)

```


## Set working directory to proper folder

```{r include=FALSE}

# set working directory to proper repo

setwd("/Users/jaredfreedman/Research/git/edna_metabarcoding/GC_VR_MiSeq_2021")

```

## Check for PhiX

```{r include=FALSE}
# check for PhiX

# only subsample 10000 reads

if(T){
  system2("vsearch", 
          "-fastx_subsample GC_VR_MiSeq_Demulti/Undetermined_S0_R1_001.fastq -fastaout GC_VR_MiSeq_Demulti/Undetermined_S0_R1_001.fastq -sample_pct 100")
} else { # check all reads in read 1 for PhiX
  system2("paste",
          " - - - - < /GC_VR_MiSeq_Demulti/Undetermined_S0_R1_001.fastq | cut -f 1,2 | sed 's/^@/>/' | tr \"\t\" \"\n\" > /GC_VR_MiSeq_Demulti/Undetermined_S0_R1_001.fastq")
}


system2("vsearch",
        "-usearch_global /GC_VR_MiSeq_Demulti/Undetermined_S0_R1_001.fastq -db /Users/jaredfreedman/Research/git/JAMP/Tutorial/PhiX.fasta -id 0.9 -strand both -blast6out PhiX_table.txt -maxrejects 1 -maxaccepts 1")
```

## Merge pair end reads

```{r}
demulti_files <- list.files("demulti_fastq/_data", full.names = TRUE)


Merge_PE(files = demulti_files)

#Successfully merged

```


## Trim primers
fwhF2 and EPTDr2n

EPTDr2n ended with a Y degenerate base, which IDT was unable to produce. Instead, two primers were created, one ending with C (EPTDr2n_C) and the other ending with T (EPTDr2n_T).

All primers are sets of 4 primers with 0-3 N bases to create frame shifts.

```{r include=FALSE}

# import primer sequences (just COI primer sequence, not including universal tail)

primers <- read.csv("Sites_and_Samples/fwhF2_EPTDr2n_primer_sequences.csv")

#identify name of forward and reverse primer
f.primer <- "fwhF2_"
r.primer <- "EPTDr2n_"

#return TRUE or FALSE if each primer is Forward or Reverse
is_f <- grepl(f.primer, primers$Name)
is_r <- grepl(r.primer, primers$Name)

#create character string with forward and reverse primers
fwhF2 <- primers$Sequence[primers$Name=="fwhF2"]
EPTDr2n <- primers$Sequence[primers$Name=="EPTDr2n"]

primer_sequences <- cbind(fwhF2,EPTDr2n)

# trim primers 
merged_files <- list.files("A_merge_PE/_data", full.names = TRUE)

Cutadapt(files = merged_files, 
         forward = fwhF2,
         reverse = EPTDr2n,
         bothsides=T)



#by using "bothsides=T", forward or reverse primers are detected on both ends. This is not nessesary for fusion primers.
```


## get length distribution of trimmed sequences

```{r include=FALSE}

dir.create("_Length_distribution", path = "C_Cutadapt/_stats/_Length_distribution")

trimmed_files <- list.files(path = "C_Cutadapt/_Data", full.names = TRUE) # Create list of file names
nfile <- length(trimmed_files)
ifile <- 1

start_time <- Sys.time() # start timer to see how long the following for loop will take

for(ifile in 1:nfile){

  file <- trimmed_files[ifile]   #create character string containing the file path and name
  filename <- strsplit(trimmed_files[ifile], split="_PE_cut.fastq")[[1]]   #create character string that contains file name without _PE_cut or .fastq extention
  filename <- strsplit(filename, split="/")[[1]][3]   #remove the file path from the filename (done by splitting string by /, and only taking the 3rd part)

  
  Length_distribution(sequFile = file,
                      out = paste("C_Cutadapt/_stats/_Length_distribution/", filename, ".pdf", sep=""),
                      fastq=TRUE)
  
  print(paste("Length distribution histogram created for ", filename, sep=""))
  
}

end_time <- Sys.time() # end timer once for loop is completed

end_time - start_time



```


## Discard non-target length reads (+/- 10bp)
The fwhF2/EPDr2n COI fragment is 142bp

```{r include=FALSE}

Cutadapt_files <- list.files("D_Cutadapt/_data", full.names = TRUE)

Minmax(files = Cutadapt_files,
       min=(142-10),
       max=(142+10))

```


## Discard reads with an Expected Error greater than 1

```{r include=FALSE}

Minmax_files <- list.files("E_Minmax/_data", full.names = TRUE)

Max_ee(files = Minmax_files,
       max_ee=1)
#Output in F_U_max_ee 

Max_ee(files = Minmax_files,
       max_ee=0.5)
#Output in G_U_max_ee

```


## Cluster OTUs 
Cluster sequences into OTUs using VSearch.
This OTU clustering is not subset to the lowest sequencing depth, something that is available with USearch but not with VSearch. At the moment I am going to cluster without subsetting, but plan to figure out how to subset in the future (11/19/21).


```{r include=FALSE}

start_time <- Sys.time() # start timer to see how long the following for loop will take

# Take files from F_U_max_ee, which were filtered at an EE of 1
Max_ee_files <- list.files("F_U_max_ee/_data", full.names=T)
Max_ee_subset <- Max_ee_files[1:10]

Cluster_otus(files= Max_ee_files, filter=0.01)



end_time <- Sys.time() # end timer once for loop is completed

end_time - start_time


```






