---
title: "JAMP Pipeline Tutorial"
author: "Jared Freedman"
date: "10/26/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# R Markdown

## Set Working Directory

```{r}
# set working directory to proper repo

setwd("/Users/jaredfreedman/Research/git/JAMP/Tutorial")
# download the raw data from and place it into the tutorial folder (Study we are looking at https://peerj.com/articles/3006/) 
# MiSeq Run1, R1 direction:
# https://dx.doi.org/10.6084/m9.figshare.4039821.v1
# MiSeq Run1, R2 direction:
# https://dx.doi.org/10.6084/m9.figshare.4039860.v1
```

## Download tutorial fastq files and set up folders (if not starting from beginning)

```{r}
library("JAMP")

# Download tutorial fastq files to wd -------------------------------------

download.file("https://ndownloader.figshare.com/files/6503952", "16_S10_L001_R1_001_run1.fastq.gz")
download.file("https://ndownloader.figshare.com/files/6503991", "16_S10_L001_R2_001_run1.fastq.gz")
system2("gunzip", "16_S10_L001_R1_001_run1.fastq.gz")
system2("gunzip", "16_S10_L001_R2_001_run1.fastq.gz")

list.files()

# JAMP generates a new folder for each processing step! Should your files alreay be demultiplexed or you want to start somwhere else in the pipeline with preprocessed reads, you can generate an empty folder and place your files to be processed in "_data"
Empty_folder()

#To delete the last generated folder, run
Remove_last_folder()
```

## Demultiplex samples

```{r}
# In this example we are dealing with sequence raw data that is not yet demultiplexed. To demultiplex run:


Demultiplexing_shifted(file1="16_S10_L001_R1_001_run1.fastq", file2="16_S10_L001_R2_001_run1.fastq", tags="_converter/indexe_1.csv", combinations="_converter/combos_1.csv")


#file1 and file2 are the sequencing files, indexe_1.csv is the index file, and combos_1.csv is the combination of shifts in primers
```

## Check for PhiX

```{r}
# check for PhiX

# only subsample 10000 reads

if(T){
  system2("vsearch", "-fastx_subsample A_Demultiplexing_shifted/_data/N_debris_R1.fastq -fastaout A_Demultiplexing_shifted/_data/N_debris_R1.fasta -sample_size 10000")
} else { # check all reads in read 1 for PhiX
  system2("paste", " - - - - < A_Demultiplexing_shifted/_data/N_debris_R1.fastq | cut -f 1,2 | sed 's/^@/>/' | tr \"\t\" \"\n\" > A_Demultiplexing_shifted/_data/N_debris_R1.fasta")
}


system2("vsearch", "-usearch_global A_Demultiplexing_shifted/_data/N_debris_R1.fasta -db PhiX.fasta -id 0.9 -strand both -blast6out PhiX_table.txt -maxrejects 1 -maxaccepts 1")
```

